src/main/java/sk/uniba/fmph/dcs/game_board/GetChoice.java

```diff
@@ -0,0 +1,28 @@
+package sk.uniba.fmph.dcs.game_board;
+
+import sk.uniba.fmph.dcs.stone_age.ActionResult;
+import sk.uniba.fmph.dcs.stone_age.Effect;
+import java.util.ArrayList;
+
+public class GetChoice implements EvaluateCivilisationCardImmediateEffect {
+    private int numberOfResources;
+
+    public GetChoice(final int numberOfResources) {
+        this.numberOfResources = numberOfResources;
+    }
+
+    @Override
+    public final ActionResult performEffect(final Player player, final Effect choice) {
+        if (!choice.isResource()) {
+            return ActionResult.FAILURE;
+        }
+        if (numberOfResources <= 0) {
+            return ActionResult.FAILURE;
+        }
+        numberOfResources--;
+        ArrayList<Effect> c = new ArrayList<>();
+        c.add(choice);
+        player.playerBoard().giveEffect(c);

```

Comment:
```suggestion
        if (numberOfResources != choice.size()) {
            return ActionResult.FAILURE;
        }
        for (Effect singleResource : choice) {
            if (!singleResource.isResource) {
                return ActionResult.FAILURE;
            }
        }
        player.playerBoard().giveEffect(choice);
        

```
Toto mne dava vacsi zmysel ked choice bude arraylist, ked karta umoznuje zobrat viac ako jeden resource hracovej volby. Asi to je Lukotkova chyba.